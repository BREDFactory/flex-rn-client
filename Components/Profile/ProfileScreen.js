// @flow
import React from "react";
import {
  Button,
  Card,
  FormInput,
  Text,
  List,
  ListItem,
  SearchBar
} from "react-native-elements";

import { View, TextInput, AsyncStorage, ScrollView } from "react-native";
import config from "../../config/api";
import server from "../../config/server";
import styles from "./ProfileScreenStyles";
import { NavigationScreenProp } from 'react-navigation'
import { filterÂ } from 'ramda';

type State = {
  name: ?string,
  fname: ?string,
  id: ?string,
  place: ?string,
  debug: ?array
};

type Props = {
  navigate: NavigationScreenProp<{}>,
};

class ProfileScreen extends React.Component<Props, State> {
  static navigationOptions = {
    title: "Profile"
  };

  constructor() {
    super();
    this.state = {
      name: "",
      fname: "",
      id: "",
      place: "",
      debug: "",
      search: "",
    };
  }

  componentDidMount() {
    AsyncStorage.getItem("USER", (err, result) => {
      if (err || result === null) this.goTo("Login");
      else {
        this.setState(JSON.parse(result));
      }
    });
  }

  /** This function is used to send a new place to the server */

  sendToServ(ctx, json) {
    if (
      ctx.state.name !== "" &&
      ctx.state.fname !== "" &&
      ctx.state.id !== "" &&
      ctx.state.place !== ""
    ) {
      ctx = ctx || window;

      let payload = {
        name: ctx.state.name,
        fname: ctx.state.fname,
        id_user: ctx.state.id,
        id_place: ctx.state.place
      };
      console.log(payload);
      fetch(server.address, {
        method: "POST",
        body: JSON.stringify(payload),
        headers: {
          "Content-Type": "application/json",
          "x-access-token": config.token
        }
      })
        .then(res => res.json())
        .then(data => {
          var redirect = true;
          json.map(
            element =>
              payload.id_place == element.id && element.using
                ? (redirect = false)
                : null
          );
          if (redirect) {
            AsyncStorage.setItem("USER", JSON.stringify(ctx.state));

            ctx.goTo("Leave");
          }
        });
    }
  }

  /** This function is used to get the places from the server */
  getPlaces(ctx, fn, l = null) {
    ctx = ctx || window;

    fetch(server.address + "places/", {
      method: "GET",
      headers: {
        "Content-Type": "application/x-www-form-urlencoded",
        "x-access-token": config.token
      }
    })
      .then(res => res.json()) // transform data to json
      .then(data => {
        if (l) {
          fn(ctx, l);
        } else {
          fn(ctx, data);
        }
      });
  }

  async setDebug(ctx, json) {
    const result = json.filter(
      element => element !== null && element.using === false
    );
    console.log(result);
    ctx.setState({ debug: result });
  }

  goTo(str) {
    const navigation = this.props.navigation;
    navigation.popToTop();
    navigation.navigate(str);
  }

  logOut() {
    AsyncStorage.removeItem("USER");
    this.goTo("Login");
  }

  _handleSearch = search => {
    this.setState({ search });
  };

  _handleList = () => {
    const { debug, search } = this.state;

    const newT =
      debug !== ""
        ? debug.filter(e => {
            let finalResult = true;
            for (let element in search) {
              if (search[element] !== e.id[element]) {
                finalResult = false;
              }
            }
            return finalResult;
          })
        : debug;
    return search === "" ? debug : newT;
  };

  /** This function is used to attach the current user to a place  */
  getUser(ctx, l) {
    if (
      ctx.state.name !== "" &&
      ctx.state.fname !== "" &&
      ctx.state.id !== "" &&
      l.id !== ""
    ) {
      ctx = ctx || window;

      let payload = {
        name: ctx.state.name,
        fname: ctx.state.fname,
        id_user: ctx.state.id,
        id_place: l.id
      };
      fetch(server.address, {
        method: "POST",
        body: JSON.stringify(payload),
        headers: {
          "Content-Type": "application/json",
          "x-access-token": config.token
        }
      })
        .then(res => res.json())
        .then(data => {
          let redirect = true;
          payload.id_place == l.id && l.using ? (redirect = false) : null;
          if (redirect) {
            AsyncStorage.setItem(
              "USER",
              JSON.stringify({
                name: payload.name,
                fname: payload.fname,
                id: payload.id_user,
                place: payload.id_place,
                debug: ctx.state.debug
              })
            );

            ctx.goTo("Leave");
          }
        });
    }
  }

  render() {
    const navigation = this.props.navigation;
    const { fname, name, id, debug } = this.state;

    return (
      <ScrollView style={styles.view}>
        <View style={styles.view_second}>
          <Text h4 style={styles.text_first}>
            {fname} {name} [{id}]
          </Text>
          <Button
            fontWeight="bold"
            fontSize={12}
            borderRadius={15}
            backgroundColor="#5167A4"
            color="#fff"
            style={styles.logOut}
            title="Log Out"
            onPress={() => this.logOut()}
          />
        </View>

        <Card title="Manual insertion">
          <FormInput
            style={styles.place}
            placeholder="Place"
            onChangeText={text => this.setState({ place: text })}
          />

          <View style={styles.sendContainer}>
            <Button
              fontWeight="bold"
              borderRadius={15}
              backgroundColor="#5167A4"
              color="#fff"
              style={styles.send}
              title="Send"
              onPress={() => this.getPlaces(this, this.sendToServ)}
            />
          </View>
        </Card>

        <Card title="Scan QR code">
          <View style={styles.scan_container}>
            <Button
              fontWeight="bold"
              borderRadius={15}
              backgroundColor="#5167A4"
              color="#fff"
              style={styles.scan}
              title="Scan"
              onPress={() => navigation.navigate("Scan")}
            />
          </View>
        </Card>
        <Card>
          <View style={styles.emptyPlaces_container}>
            <Button
              fontWeight="bold"
              large={false}
              borderRadius={15}
              backgroundColor="#5167A4"
              color="#fff"
              style={styles.free_places}
              title="Free places"
              onPress={() => this.getPlaces(this, this.setDebug)}
            />
          </View>
          <SearchBar
            onChangeText={this._handleSearch}
            round
            lightTheme
            platform="ios"
            containerStyle={{
              backgroundColor: "white",
              borderWidth: 0,
              marginTop: 10
            }}
            searchIcon={{ size: 24 }}
            placeholder="Search a place..."
          />
          {debug !== "" && debug ? (
            <List containerStyle={{ marginBottom: 20 }}>
              {this._handleList().map(
                item =>
                  item ? (
                    <ListItem
                      onPress={() => this.getPlaces(this, this.getUser, item)}
                      key={item.id}
                      title={item.id}
                    />
                  ) : null
              )}
            </List>
          ) : null}
        </Card>
      </ScrollView>
    );
  }
}

export default ProfileScreen;
