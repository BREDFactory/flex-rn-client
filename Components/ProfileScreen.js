import React from 'react'

import { View, Text, Button, TextInput } from 'react-native'
import { AsyncStorage } from 'react-native';
import config from '../config/api';
import server from '../config/server';


class ProfileScreen extends React.Component {
  static navigationOptions = {
    title: 'Profile',
  };

  constructor(){
    super();
    this.state = {
      name: '',
      fname: '',
      id:'',
      place:'',
      debug:'',
    }
  }

  componentDidMount(){
    AsyncStorage.getItem('USER', (err, result) => {
      if(err || result == null)
        this.goTo('Login');
      else {
        this.setState(JSON.parse(result));
      }
    });
  }

  sendToServ(ctx, json) {
    if(ctx.state.name != '' && ctx.state.fname != '' && ctx.state.id != '' && ctx.state.place != '')
    {
      var ctx = ctx || window;

      var payload = {
          name: ctx.state.name,
          fname: ctx.state.fname,
          id_user: ctx.state.id,
          id_place: ctx.state.place
      };

      fetch(server.address , {
        method: "POST",
        body: JSON.stringify(payload),
        headers: {
          "Content-Type": "application/json",
          "x-access-token": config.token
        }
      })
      .then(function(res){
        return res.json();
      })
      .then(function(data){
        var redirect = true;
        json.forEach(function(element){
          if(payload.id_place == element.id && element.using)
            redirect = false;
        });
        if(redirect)
        {
          ctx.state.debug = 'Wellcome';
          AsyncStorage.setItem('USER', JSON.stringify(ctx.state));

          ctx.goTo('Leave');
        }
      })
    }
  }

  getPlaces(ctx, fn) {
    ctx = ctx || window;

    fetch(server.address + "places/", {
      method: "GET",
      headers: {
        "Content-Type": "application/x-www-form-urlencoded",
        "x-access-token": config.token
      }
    })
    .then(function(res){ return res.json(); })//transform data to json
    .then(function(data)
    {
      fn(ctx, data);
    });
  }

  async setDebug(ctx, json) {
    var result = '';
    json.forEach(function(element){
      if(!element.using)
      {
        result += element.id + ',';
      }
    });
    ctx.setState({debug:result});
  }

  goTo(str) {
    const navigation = this.props.navigation;
    navigation.popToTop();
    navigation.navigate(str);
  }

  _logOut() {
    AsyncStorage.removeItem('USER');
    this.goTo('Login');
  }

  render() {
    const navigation = this.props.navigation;

    return (
      <View style={{flexDirection:'column',}}>

        <View style={{flexDirection: 'row', justifyContent: 'space-between',margin:5}}>
          <Text style={{margin:5, marginTop:8}}>{this.state.fname} {this.state.name} [{this.state.id}]</Text>

          <Button title='Log Out'
          color="#000"
          onPress={() => this._logOut()}/>
        </View>

        <View style={{ borderRadius: 4, borderWidth: 0.5,}}/>

        <Text style={{marginLeft:10,marginTop:20}}>Insertion manuelle</Text>

        <View style={{margin:10, borderRadius: 4, borderWidth: 0.5,marginRight:50,marginLeft:50}}>
          <TextInput
            placeholder="Place"
            onChangeText={(text) => this.setState({place:text})}
          />
        </View>

        <View style={{marginRight:50,marginLeft:50,}}>
          <Button title='Send'
          color="#000"
          onPress={() => this.getPlaces(this, this.sendToServ)}/>
        </View>

        <View style={{ marginTop:20, borderRadius: 4, borderWidth: 0.5,}}/>

        <Text style={{marginLeft:10,marginTop:20}}>Scan QR code</Text>

        <View style={{margin:20, marginRight:50, marginLeft:50, justifyContent:'flex-end'}}>
          <Button title='Scan'
          color="#000"
          onPress={() => navigation.navigate('Scan') }/>
        </View>

        <View style={{ marginTop:20, borderRadius: 4, borderWidth: 0.5,}}/>

        <View style={{marginTop:20,marginLeft:50, marginRight:50}}>
          <Button title='Places libres'
          color="#000"
          onPress={() => this.getPlaces(this, this.setDebug)}/>
        </View>
        <Text>{ this.state.debug }</Text>

      </View>
    )
  }
}

export default ProfileScreen;
