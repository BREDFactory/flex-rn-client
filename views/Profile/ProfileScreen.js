/*
Copyright 2019-2020 BRED Banque Populaire

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/
// @flow
/* eslint-disable */
import React from "react";

import { AsyncStorage, ScrollView, View, Text, Alert } from "react-native";
import socketIOClient from "socket.io-client";

import LinearGradient from "react-native-linear-gradient";
import { NavigationScreenProp, NavigationEvents } from "react-navigation";
import Icon from "react-native-vector-icons/FontAwesome";
import config from "../../config/api";
import server from "../../config/server";
import regex from "../../config/regex";
import styles from "./ProfileScreenStyles";
import { getPlaces, goTo, sendToServ, leavePlace } from "../../utils/utils";
import I18n from "../../i18n/i18n";

import LottieView from "lottie-react-native";

/**
 * List of components
 */
import ManualInsertionCard from "@components/Profile/components/ManualInsertionCard";
import HeaderCard from "@components/Profile/components/HeaderCard";
import QRCodeComponent from "@components/Profile/components/QRCodeComponent";
import LeaveButton from "@components/Leave/LeaveButton";

type Historical = {
  place_id: string,
  begin: string,
  end: string
};

type State = {
  name: string,
  fname: string,
  id: string,
  place: string,
  historical: Array<Historical>,
  debug: Array<any> | string,
  isWrongFormatPlace: boolean,
  placeTaken: boolean
};

type Props = {
  navigation: NavigationScreenProp<{}>
};

class ProfileScreen extends React.Component<Props, State> {
  static navigationOptions = ({ navigation }) => {
    return {
      title: I18n.t("profile.title"),
      tabBarIcon: ({ tintColor }) => (
        <Icon name="qrcode" size={23} color={tintColor} />
      )
    };
  };

  constructor() {
    super();
    this.placeInput = "";
    this.socket = socketIOClient(server.sockets);
    this.socket.on('leavePlace', () => this.leavePlace(this));
    this.state = {
      name: "",
      fname: "",
      id: "",
      place: "",
      isWrongFormatPlace: false,
      placeTaken: false
      // progress: new Animated.Value(0),
    };
  }

  componentDidMount = () => {
    const { navigation } = this.props;
    AsyncStorage.getItem("USER", (err, result) => {
      if (err || result === null) goTo(this, "Login");
      else {
        result = JSON.parse(result);
        if (result.placeTaken)
          this.socket.emit('checkPlace', result.place);
        this.setState(result);
        navigation.setParams(result);
        const userId = result.id;
        fetch(`${server.address}users/${userId}`, {
          method: "GET",
          headers: {
            "Content-Type": "application/x-www-form-urlencoded",
            "x-access-token": config.token
          }
        })
          .then(res => res.json()) // transform data to json
          .then(data => {
            this.setState({ historical: data.historical || [] });
          });
      }
    });
  };

  onSuccess = async e => {
    if (e.data.match(regex.place_regex) !== null) {
      this.setState({ place: e.data });
      getPlaces(this, sendToServ);
    } else {
      this.setState({ isWrongFormatPlace: true });
    }
  };

  DefaultComponent = () => {
    const {
      fname,
      name,
      id,
      isWrongFormatPlace,
    } = this.state;

    insertPlace = () => {
      if (this.placeInput !== "" && this.placeInput.match(regex.place_regex) !== null) {
        const payload = {
          id_user: id,
          id_place: this.placeInput
        };

        fetch(`${server.address}/take_place`, {
          method: "POST",
          headers: {
            "Content-Type": "application/json",
            "x-access-token": config.token
          },
          body: JSON.stringify(payload)
        })
          .then(res => {
            if (res.status === 200) {
              this.setState({
                place: this.placeInput,
                placeTaken: true
              });
              this.socket.emit('joinRoom', this.placeInput);
              AsyncStorage.setItem("USER", JSON.stringify(this.state))
            }
            else if (res.status === 500) {
              res.json().then(user => {
                Alert.alert("Impossible", `Place déjà utilisée par : ${user.fname} ${user.name}`);
              })
            }
          });
      } else this.setState({ isWrongFormatPlace: true });
    }

    return (
      <ScrollView style={styles.view}>
        <HeaderCard fname={fname} name={name} id={id} />
        <View>
          <QRCodeComponent onRead={this.onSuccess} />
          <View>
            <ManualInsertionCard
              onChangeText={text => this.placeInput = text}
              onSubmitEditing={() => insertPlace()}
              onPress={() => insertPlace()}
            />
            {isWrongFormatPlace ? (
              <Text style={styles.debug}>{I18n.t("profile.format")}</Text>
            ) : null}
          </View>
        </View>
      </ScrollView>
    );
  };

  LeaveComponent = () => {
    const { place } = this.state;
    // Animated.timing(this.state.progress, {
    //   toValue: 1,
    //   duration: 7000
    // }).start();

    return (
      <View
        style={{
          backgroundColor: "white",
          flex: 1,
          justifyContent: "center",
          alignItems: "center"
        }}
      >
        {/* <LottieView
          source={require("./animation.json")}
          progress={this.state.progress}
        /> */}
        <LeaveButton place={place} onPress={() => this.leavePlace(this)} />
      </View>
    );
  };

  Content = ({ place }) => {
    if (!place) {
      return <this.DefaultComponent />;
    }
    return <this.LeaveComponent />;
  };

  async leavePlace(ctx) {
    const { id } = this.state;
    ctx = ctx || window;
    await fetch(`${server.address}users/${id}`, {
      method: "GET",
      headers: {
        "Content-Type": "application/x-www-form-urlencoded",
        "x-access-token": config.token
      }
    })
      .then(res => res.json()) // transform data to json
      .then(data => {
        ctx.setState({ historical: data.historical });
      });

    const { name, fname, place, historical, remoteDay } = this.state;

    const payload: Payload = {
      name,
      fname,
      id_user: id,
      id_place: place,
      historical,
      remoteDay
    };

    await fetch(server.address, {
      method: "POST",
      body: JSON.stringify(payload),
      headers: {
        "Content-Type": "application/json",
        "x-access-token": config.token
      }
    })
      .then(res => {
        if (res.ok) {
          return res.json();
        }
        ctx.setState({ debug: "ERROR" });
      })
      .then(data => {
        ctx.setState({
          placeTaken: false,
          isWrongFormatPlace: false,
          place: "",
          debug: ""
        });
        AsyncStorage.setItem("USER", JSON.stringify(this.state));
        this.socket.emit('leaveRoom', '31V');
      });
  }

  render() {
    const { placeTaken } = this.state;

    return <this.Content place={placeTaken} />;
  }
}

export default ProfileScreen;
