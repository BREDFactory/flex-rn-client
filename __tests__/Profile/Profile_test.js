import 'react-native';
import React from 'react';
import ProfileScreen from '../../Components/Profile/ProfileScreen';

import renderer from 'react-test-renderer';

const navigation = { navigate: jest.fn(), popToTop: jest.fn() };

it('renders correctly', () => {
  const tree = renderer.create(
    <ProfileScreen navigation={navigation} />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});