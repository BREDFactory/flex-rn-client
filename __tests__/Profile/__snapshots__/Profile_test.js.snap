// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            Object {
              "fontSize": 20,
              "margin": 5,
              "marginTop": 8,
            },
          ]
        }
      >
        
         
        
         [
        
        ]
      </Text>
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            undefined,
            Object {
              "borderRadius": 15,
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={Object {}}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#5167A4",
                },
                Object {
                  "borderRadius": 15,
                },
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    Object {
                      "color": "#fff",
                    },
                    Object {
                      "fontSize": 17.5,
                    },
                    Object {
                      "fontSize": 12,
                    },
                    undefined,
                    Object {
                      "fontWeight": "bold",
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Log Out
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#e1e8ee",
            "borderWidth": 1,
            "margin": 15,
            "marginBottom": 0,
            "padding": 15,
            "shadowColor": "rgba(0,0,0, .2)",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
            undefined,
          ]
        }
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "#43484d",
                    "fontSize": 17.5,
                    "fontWeight": "bold",
                    "marginBottom": 15,
                    "textAlign": "center",
                  },
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Manual insertion
          </Text>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#e1e8ee",
                  "height": 1,
                },
                Array [
                  Object {
                    "marginBottom": 15,
                  },
                  undefined,
                ],
              ]
            }
          />
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#bdc6cf",
              "borderBottomWidth": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Place"
            style={
              Array [
                Object {
                  "color": "#86939e",
                  "minHeight": 36,
                  "width": 750,
                },
                Object {
                  "fontSize": 17.5,
                },
                undefined,
              ]
            }
          />
        </View>
        <View
          style={
            Object {
              "marginLeft": 50,
              "marginRight": 50,
              "marginTop": 20,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 15,
                  "marginRight": 15,
                },
                undefined,
                undefined,
                Object {
                  "borderRadius": 15,
                },
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={Object {}}
            >
              <View
                pointerEvents="box-only"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#9E9E9E",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 19,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "backgroundColor": "#5167A4",
                    },
                    Object {
                      "borderRadius": 15,
                    },
                    Object {
                      "padding": 12,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 20,
                        },
                        Object {
                          "color": "#fff",
                        },
                        Object {
                          "fontSize": 17.5,
                        },
                        undefined,
                        undefined,
                        Object {
                          "fontWeight": "bold",
                        },
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  Send
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#e1e8ee",
            "borderWidth": 1,
            "margin": 15,
            "marginBottom": 0,
            "padding": 15,
            "shadowColor": "rgba(0,0,0, .2)",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
            undefined,
          ]
        }
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "#43484d",
                    "fontSize": 17.5,
                    "fontWeight": "bold",
                    "marginBottom": 15,
                    "textAlign": "center",
                  },
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Scan QR code
          </Text>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#e1e8ee",
                  "height": 1,
                },
                Array [
                  Object {
                    "marginBottom": 15,
                  },
                  undefined,
                ],
              ]
            }
          />
        </View>
        <View
          style={
            Object {
              "justifyContent": "flex-end",
              "margin": 20,
              "marginLeft": 50,
              "marginRight": 50,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 15,
                  "marginRight": 15,
                },
                undefined,
                undefined,
                Object {
                  "borderRadius": 15,
                },
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={Object {}}
            >
              <View
                pointerEvents="box-only"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#9E9E9E",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 19,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "backgroundColor": "#5167A4",
                    },
                    Object {
                      "borderRadius": 15,
                    },
                    Object {
                      "padding": 12,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 20,
                        },
                        Object {
                          "color": "#fff",
                        },
                        Object {
                          "fontSize": 17.5,
                        },
                        undefined,
                        undefined,
                        Object {
                          "fontWeight": "bold",
                        },
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  Scan
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#e1e8ee",
            "borderWidth": 1,
            "margin": 15,
            "marginBottom": 0,
            "padding": 15,
            "shadowColor": "rgba(0,0,0, .2)",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "marginLeft": 50,
              "marginRight": 50,
              "marginTop": 20,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 15,
                  "marginRight": 15,
                },
                undefined,
                undefined,
                Object {
                  "borderRadius": 15,
                },
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={Object {}}
            >
              <View
                pointerEvents="box-only"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#9E9E9E",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 19,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "backgroundColor": "#5167A4",
                    },
                    Object {
                      "borderRadius": 15,
                    },
                    Object {
                      "padding": 12,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 20,
                        },
                        Object {
                          "color": "#fff",
                        },
                        Object {
                          "fontSize": 17.5,
                        },
                        undefined,
                        undefined,
                        Object {
                          "fontWeight": "bold",
                        },
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  Free places
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#393e42",
                "borderBottomColor": "#000",
                "borderBottomWidth": 1,
                "borderTopColor": "#000",
                "borderTopWidth": 1,
              },
              Object {
                "backgroundColor": "#e1e8ee",
                "borderBottomColor": "#e1e1e1",
                "borderTopColor": "#e1e1e1",
              },
              Object {
                "backgroundColor": "white",
                "borderWidth": 0,
                "marginTop": 10,
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Search a place..."
            placeholderTextColor="#86939e"
            platform="ios"
            searchIcon={
              Object {
                "size": 24,
              }
            }
            style={
              Array [
                Object {
                  "backgroundColor": "#303337",
                  "borderRadius": 3,
                  "color": "#86939e",
                  "fontSize": 17.5,
                  "height": 30,
                  "margin": 8,
                  "overflow": "hidden",
                  "paddingLeft": 26,
                  "paddingRight": 19,
                },
                Object {
                  "backgroundColor": "#bdc6cf",
                },
                false,
                Object {
                  "borderRadius": 15,
                },
                undefined,
                undefined,
                false,
              ]
            }
            underlineColorAndroid="transparent"
          />
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#86939e",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "position": "absolute",
                    "top": 15.5,
                  },
                  Object {
                    "left": 16,
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
